lib/geomancer.ex

```diff
@@ -1,21 +1,19 @@
 defmodule Geomancer do
   @moduledoc """
-  A library to convert between various geospatial file formats
+  A behaviour to convert geospatial data from one format to another.
   """
 
-  alias Geomancer.Shapefile
+  @type input_path() :: String.t()
+  @type conversion() :: String.t()
+  @type reason() :: String.t()
+  @type geo_struct() :: struct | map
 
-  @type geo_json :: String.t()
-  @type reason :: String.t()
+  @callback from(input_path) :: {:ok, conversion} | {:error, reason}
+  @callback read(input_path) :: {:ok, geo_struct} | {:error, reason}
 
-  @spec geo_json(String.t()) :: {:ok, geo_json()} | {:error, reason()}
-  def geo_json(input_filepath) do
-    case Path.extname(input_filepath) do
-      ".zip" ->
-        Shapefile.geo_json(input_filepath)
-
-      ext ->
-        {:error, "Unsupported format: #{ext}"}
+  defmacro __using__(_) do
+    quote do
+      @behaviour Geomancer
```

Comment:
I would rather the @behaviour be added explicitly to geo_json and shapefile unless the using is going to contain more in the future.

Reply:
I believe it is, but let me think through that.